synfig-studio/src/gui/modules/mod_palette/dock_paledit.cpp

```diff
@@ -349,6 +369,68 @@ Dock_PalEdit::add_color(const synfig::Color& x)
 	return size()-1;
 }
 
+void
+Dock_PalEdit::add_from_clipboard()
+{
+	if(clipboard_==NULL)
```

Comment:
Maybe use `Gdk::Clipboard` here?
https://developer-old.gnome.org/gtkmm/stable/classGdk_1_1Clipboard.html#details

Reply:
According to the link that was posted I need to have the following library:
```#include <gdkmm/clipboard.h>```
however when I try to compile with adding that include it gives me the following error: 
``` fatal error: gdkmm/clipboard.h: No such file or directory```
  ``` 34 | #include <gdkmm/clipboard.h> ```
   
it seems like that part of gdk at least is not part of the synfig dependencies?

Reply:
Sorry I gave you the wrong link. Missed this part: 

> [Since gtkmm 3.94:](https://developer-old.gnome.org/gtkmm/stable/since_3_94.html#_since_3_94000002)

So this link is for GTK4.
Found only example for GTK3: https://developer-old.gnome.org/gtkmm-tutorial/3.24/sec-clipboard-examples.html.en

But also I found this comment:
https://gitlab.gnome.org/GNOME/gtkmm/-/issues/110

> I will not continue forever with maintenance of gtkmm and other *mm modules.
> Unless someone else continues the work, glibmm, gtkmm and other modules will
> be abandoned in a near future.

So now I'm not sure if we should prefer gtkmm to gtk. 

@rodolforg @morevnaproject Any ideas here?

Reply:
I've ended up sticking with the functions described in this page 
https://developer-old.gnome.org/gtkmm/3.24/classGtk_1_1Clipboard.html 
which was posted by rodolforg.